From 90595daa5cab2d154f0b4dd7ab5b8c8a466752ec Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan.agner@toradex.com>
Date: Thu, 16 Feb 2017 17:06:24 -0800
Subject: [PATCH] ARM: dts: imx7-colibri: introduce low power carrier board

Introduce Low Power Demo (lpd) carrier board. The carrier board is
built to show i.MX 7 heterogeneous architecture capabilities. It
makes use of the M4 core to control a LCD display and several
sensors. It also can run from battery and has charging capabilities.

Additionally, the board uses the ublox EMMY-W1 (W165) WiFi solution.

Signed-off-by: Stefan Agner <stefan.agner@toradex.com>
---
 arch/arm/boot/dts/Makefile              |   2 +
 arch/arm/boot/dts/imx7-colibri-lpd.dtsi | 321 ++++++++++++++++++++++++++++++++
 arch/arm/boot/dts/imx7d-colibri-lpd.dts |  23 +++
 arch/arm/boot/dts/imx7s-colibri-lpd.dts |  18 ++
 4 files changed, 364 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx7-colibri-lpd.dtsi
 create mode 100644 arch/arm/boot/dts/imx7d-colibri-lpd.dts
 create mode 100644 arch/arm/boot/dts/imx7s-colibri-lpd.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 6ad084d47853..4945438498aa 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -422,6 +422,8 @@ dtb-$(CONFIG_SOC_IMX7D) += \
 	imx7d-19x19-lpddr2-arm2.dtb \
 	imx7d-colibri-eval-v3.dtb \
 	imx7s-colibri-eval-v3.dtb \
+	imx7d-colibri-lpd.dtb \
+	imx7s-colibri-lpd.dtb \
 	imx7d-sdb.dtb \
 	imx7d-sdb-epdc.dtb \
 	imx7d-sdb-gpmi-weim.dtb \
diff --git a/arch/arm/boot/dts/imx7-colibri-lpd.dtsi b/arch/arm/boot/dts/imx7-colibri-lpd.dtsi
new file mode 100644
index 000000000000..b9bd8be771e6
--- /dev/null
+++ b/arch/arm/boot/dts/imx7-colibri-lpd.dtsi
@@ -0,0 +1,321 @@
+/*
+ * Copyright 2017 Toradex AG
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+/ {
+	chosen {
+		bootargs = "console=ttymxc0,115200";
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_pciewakeup>;
+
+		power {
+			label = "PCIe Wake-Up";
+			gpios = <&gpio3 25 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_WAKEUP>;
+			debounce-interval = <10>;
+			gpio-key,wakeup;
+		};
+	};
+
+	extcon_usbc_det: usbc_det {
+		compatible = "linux,extcon-usb-gpio";
+		debounce = <25>;
+		id-gpio = <&gpio7 14 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_usbc_det>;
+	};
+
+	reg_3v3_a7: regulator-3v3-a7 {
+		compatible = "regulator-fixed";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_en_33v_a7>;
+		regulator-name = "3.3V_A7";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		/* We can not specify all consumer, go safe and leave it on */
+		regulator-always-on;
+		gpio = <&gpio2 10 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
+	reg_3v3_wifi: regulator-3v3-wifi {
+		compatible = "regulator-fixed";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_en_33v_wifi>;
+		regulator-name = "3.3V_WiFi";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&gpio2 25 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
+	reg_5v0: regulator-5v0 {
+		compatible = "regulator-fixed";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_en_5v_a7>;
+		regulator-name = "5V_A7";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		gpio = <&gpio5 2 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
+	reg_usbh_vbus: regulator-usbh-vbus {
+		compatible = "regulator-fixed";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_usbh1_reg>;
+		regulator-name = "VCC_USB[1-4]";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+		/* Not usable since GPIO4 is used by M4! */
+		/*gpio = <&gpio4 7 GPIO_ACTIVE_LOW>;*/
+		vin-supply = <&reg_5v0>;
+	};
+};
+
+&bl {
+	brightness-levels = <0 4 8 16 32 64 128 255>;
+	default-brightness-level = <6>;
+	status = "okay";
+};
+
+&adc1 {
+	status = "okay";
+};
+
+&adc2 {
+	status = "okay";
+};
+
+&epxp {
+	status = "okay";
+};
+
+&ecspi3 {
+	status = "disabled";
+};
+
+&fec1 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "disabled";
+};
+
+&gpio4 {
+	status = "disabled";
+};
+
+&i2c4 {
+	status = "okay";
+
+	polytouch: ft5x06@38 {
+		compatible = "edt,edt-ft5406", "edt,edt-ft5x06";
+		reg = <0x38>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_ft5x06>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <23 IRQ_TYPE_EDGE_FALLING>;
+		reset-gpios = <&gpio5 11 GPIO_ACTIVE_LOW>;
+	};
+
+	/* M41T0M6 real time clock on carrier board */
+	rtc: m41t0m6@68 {
+		compatible = "st,m41t00";
+		reg = <0x68>;
+	};
+};
+
+&lcdif {
+	display = <&display0>;
+	status = "okay";
+
+	display0: lcd-display {
+		bits-per-pixel = <16>;
+		bus-width = <18>;
+
+		display-timings {
+			native-mode = <&timing_vga>;
+
+			/* Standard VGA timing */
+			timing_vga: 640x480 {
+				clock-frequency = <25175000>;
+				hactive = <640>;
+				vactive = <480>;
+				hback-porch = <40>;
+				hfront-porch = <24>;
+				vback-porch = <32>;
+				vfront-porch = <11>;
+				hsync-len = <96>;
+				vsync-len = <2>;
+
+				de-active = <1>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				pixelclk-active = <0>;
+			};
+
+			/* WVGA Timing, e.g. EDT ET070080DH6 */
+			timing_wvga: 800x480 {
+				clock-frequency = <33260000>;
+				hactive = <800>;
+				vactive = <480>;
+				hback-porch = <216>;
+				hfront-porch = <40>;
+				vback-porch = <35>;
+				vfront-porch = <10>;
+				hsync-len = <128>;
+				vsync-len = <2>;
+
+				de-active = <1>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				pixelclk-active = <0>;
+			};
+			/* WVGA Timing, TouchRevolution Fusion 7" */
+			timing_wvga2: 800x480pixclkact {
+				clock-frequency = <33260000>;
+				hactive = <800>;
+				vactive = <480>;
+				hback-porch = <216>;
+				hfront-porch = <40>;
+				vback-porch = <35>;
+				vfront-porch = <10>;
+				hsync-len = <128>;
+				vsync-len = <2>;
+
+				de-active = <1>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				pixelclk-active = <1>;
+			};
+		};
+	};
+};
+
+&pwm1 {
+	status = "okay";
+};
+
+&pwm2 {
+	status = "disabled";
+};
+
+&pwm3 {
+	status = "disabled";
+};
+
+&pwm4 {
+	status = "disabled";
+};
+
+&uart1 {
+	pinctrl-0 = <&pinctrl_uart1>;
+	status = "okay";
+};
+
+&uart2 {
+	/* Reserved for Cortex-M4 */
+	status = "disabled";
+};
+
+&uart3 {
+	status = "okay";
+};
+
+&usbotg1 {
+	extcon = <&extcon_usbc_det>;
+	vbus-supply = <&reg_usbh_vbus>;
+	status = "okay";
+};
+
+&usdhc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc1 &pinctrl_cd_usdhc1>;
+	vmmc-supply = <&reg_3v3_wifi>;
+	no-1-8-v;
+	disable-wp;
+	non-removable;
+	enable-sdio-wakeup;
+	/*keep-power-in-suspend;*/
+
+	status = "okay";
+};
+
+&usdhc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc2 &pinctrl_cd_usdhc2>;
+	vmmc-supply = <&reg_3v3_a7>;
+	no-1-8-v;
+	disable-wp;
+	/*keep-power-in-suspend;*/
+
+	status = "okay";
+};
+
+&iomuxc {
+	pinctrl-names = "";
+	pinctrl-0 = <>;
+
+	imx7d-colibri {
+		pinctrl_en_33v_a7: en33vgpio {
+			fsl,pins = <
+				MX7D_PAD_EPDC_DATA10__GPIO2_IO10	0x74 /* SODIMM 105 */
+			>;
+		};
+
+		pinctrl_en_33v_wifi: en33vwifigpio {
+			fsl,pins = <
+				MX7D_PAD_EPDC_GDOE__GPIO2_IO25		0x14 /* SODIMM 134 */
+			>;
+		};
+
+		pinctrl_en_5v_a7: en5va7gpio {
+			fsl,pins = <
+				MX7D_PAD_SD1_RESET_B__GPIO5_IO2		0X14 /* SODIMM 73 */
+			>;
+		};
+
+		pinctrl_ft5x06: ft5x06gpio {
+			fsl,pins = <
+				MX7D_PAD_SD2_RESET_B__GPIO5_IO11	0x14 /* SODIMM 98 */
+				MX7D_PAD_LCD_DATA18__GPIO3_IO23		0x14 /* SODIMM 136 */
+			>;
+		};
+
+		pinctrl_pciewakeup: pciewakeup {
+			fsl,pins = <
+				MX7D_PAD_LCD_DATA20__GPIO3_IO25		0x14 /* SODIMM 140 */
+			>;
+		};
+
+		pinctrl_usdhc2: usdhc2grp {
+			fsl,pins = <
+				MX7D_PAD_SD2_CMD__SD2_CMD	0x59 /* SODIMM 186 */
+				MX7D_PAD_SD2_CLK__SD2_CLK	0x19 /* SODIMM 184 */
+
+				MX7D_PAD_SD2_DATA0__SD2_DATA0	0x59 /* SODIMM 23 */
+				MX7D_PAD_SD2_DATA1__SD2_DATA1	0x59 /* SODIMM 31 */
+				MX7D_PAD_SD2_DATA2__SD2_DATA2	0x59 /* SODIMM 100 */
+				MX7D_PAD_SD2_DATA3__SD2_DATA3	0x59 /* SODIMM 102 */
+			>;
+		};
+
+		pinctrl_cd_usdhc2: usdhc2-cd-grp {
+			fsl,pins = <
+				MX7D_PAD_LCD_DATA19__GPIO3_IO24	0x59 /* SODIMM 138 */
+			>;
+		};
+
+	};
+};
diff --git a/arch/arm/boot/dts/imx7d-colibri-lpd.dts b/arch/arm/boot/dts/imx7d-colibri-lpd.dts
new file mode 100644
index 000000000000..18d9c3eee1a7
--- /dev/null
+++ b/arch/arm/boot/dts/imx7d-colibri-lpd.dts
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2016 Toradex AG
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+/dts-v1/;
+#include "imx7d-colibri.dtsi"
+#include "imx7-colibri-lpd.dtsi"
+
+/ {
+	model = "Toradex Colibri iMX7D on Colibri Evaluation Board V3";
+	compatible = "toradex,colibri_imx7d-eval", "toradex,colibri_imx7d", \
+		     "fsl,imx7d";
+};
+
+&usbotg2 {
+	vbus-supply = <&reg_usbh_vbus>;
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx7s-colibri-lpd.dts b/arch/arm/boot/dts/imx7s-colibri-lpd.dts
new file mode 100644
index 000000000000..e0ffd5edef66
--- /dev/null
+++ b/arch/arm/boot/dts/imx7s-colibri-lpd.dts
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2016 Toradex AG
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+/dts-v1/;
+#include "imx7s-colibri.dtsi"
+#include "imx7-colibri-lpd.dtsi"
+
+/ {
+	model = "Toradex Colibri iMX7S on Colibri Evaluation Board V3";
+	compatible = "toradex,colibri_imx7s-eval", "toradex,colibri_imx7s", \
+		     "fsl,imx7s", "fsl,imx7d";
+};
-- 
2.12.0

